### ISSUE TITLE
Public braintree XML API

### ISSUE THREAD

**Issue by KarelJakubec** [highfidelity/fake_braintree#113]
Hi,
I would like to make something similar like fake_braintree for unit testing in python using httprety to intercept calls to BT servers.

The problem is that I have not found any documentation to BT XML API. There is no mention about message format etc. on their pages and support told me that they cannot share any doc regarding the API with me. I was able to reverse-engineer some basics from public clients, but it is slow and quite hard to determine what is really allowed to send and what not.

So...don't you posses any documents regarding the Braintree XML API that you would be able to share with rest of the world? :) Or just the source that you once used for getting this knowledge? I would be really grateful for anything, it does not have to be 100% up to date or so...

Thanks, best regards
Karel

**Comment by wacii** [comment]
The creators of this gem are no longer around. I added support for the new javascript api, but there was no documentation to help with that process, only source code and inspecting actual messages.

What's more, an upgrade to the client broke support for the drop-in ui awhile ago, and I wasn't able to figure out why. Not a lot of help I can offer.

**Comment by KarelJakubec** [comment]
Thanks for reply. 

I modified the braintree client to output all communication which helped a bit in the way I was able to mock the stuff I needed and tests are passing. Protocol is a bit strange and complex, it exchanges large xml documents with lot of duplicated data even for quite simple requests. I was unable to find there any sort of logic or architecture, maybe it is the reason for no documentation... :)

I am closing the issue as there is nothing to solve anymore. Thanks. :)
